### ISSUE TITLE
LSTM Example has same val and test set

### ISSUE THREAD

**Issue by evanmiltenburg** [fchollet/keras#3196]
The validation data and test data in [this example](https://github.com/fchollet/keras/blob/master/examples/imdb_lstm.py) are the same. It's probably better to have separate val and test data.

Or maybe I misunderstand the purpose of validation data here, and it's just to see how the model improves on the test data. As long as you don't use the val results to select the best model, and as long as you don't train multiple models to look for the best model, then maybe it's OK?

**Comment by talpay** [comment]
Correct, you're OK as long as you don't optimize hyperparameters on it. This has been brought up before in #1753. Unfortunately, no common ground was reached and the examples remain a potential trap for inexperienced beginners.
